### DISCUSSION TITLE
Idea of Split Schemas for Time Series

### DISCUSSION THREAD

**Discussion by BradKML** [alan-turing-institute/sktime#3571] [General]
1. [SlidingWindowSplitter](https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.model_selection.SlidingWindowSplitter.html) and [ExpandingWindowSplitter](https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.model_selection.ExpandingWindowSplitter.html) seemed to have other names in other documentations, namely "Walk Forward" and "Anchored Walk Forward", could this be addressed in documentation?
2. Are these types of split a kind of cross-validation (testing separate from validation)? Are there code examples of last-split validation vs extended "foresting horizon" per split? https://stats.stackexchange.com/questions/453386/working-with-time-series-data-splitting-the-dataset-and-putting-the-model-into
3. How can "N step ahead" (or whatever time units) validation testing be done, and is this done to avoid overfitting ("tuning")? Currently "forecasting horizon" as `fh` us used but it is not necessarily regular https://otexts.com/fpp3/tscv.html https://uc-r.github.io/ts_benchmarking

**Comment by topher-lo** [comment]
1. @BrandonKMLee Could you elaborate on what you mean by "other documentations"? Because "sliding window" and expanding window" seems to be the most commonly accepted nomenclature (e.g. https://www.uber.com/blog/forecasting-introduction/, https://stackoverflow.com/questions/59854723/backtesting-which-is-better-sliding-window-or-expanding-window)
2. The operation of "splitting" is not equivalent to "cross-validation". You can split a time-series dataset for many different use-cases: e.g. backtesting, to compute conformal prediction intervals, bootstrap etc. You then use these splits to perform cross-validation (i.e. split, fit, predict, score).

I think the `evaluate` function is what you are looking for https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.model_evaluation.evaluate.html

**Reply by BradKML** [reply]
1. Maybe for documentation, note them as "alternative" terms, since it has references in secondary educational materials like [Here](https://audhiaprilliant.medium.com/walk-forward-optimization-cross-validation-technique-for-time-series-data-61739f58f2c0) and [Here](https://zorro-project.com/manual/en/numwfocycles.htm) and [Here](https://robotwealth.com/rolling-and-expanding-windows-for-dummies/).
2. Thanks for bringing in `evaluate` as it is a good use case for both backtesting and "live" performance evaluations. Mind if I ask how are CIs made with splitting, since not every model has it built-in, and ensembles of models may be out-of-scope for some?

**Reply by topher-lo** [reply]
@BrandonKMLee If the model supports probabilistic intervals, then `evaluate` should work out of the box (you just need to pass in an approxiate metric e.g. pinball loss). See this line in the [`evaluate`](https://github.com/sktime/sktime/blob/v0.13.4/sktime/forecasting/model_evaluation/_functions.py#L162) implementation. What happens is within evaluate it automatically checks what "predict" method you need given the inputted `scoring`.

As for not all models having pred_interval build in...we are working hard on this! Perhaps you are interested in taking this up 🙌 https://github.com/sktime/sktime/issues/3438

Otherwise, you can trivially add probablistic intervals to ANY forecaster by passing that forecaster class into one of the following:
- https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.compose.BaggingForecaster.html
- https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.naive.NaiveVariance.html
- https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.conformal.ConformalIntervals.html

**Reply by topher-lo** [reply]
P.S. I just noticed that the probablistic metrics aren't in the API docs...but you can find them here (https://github.com/sktime/sktime/tree/main/sktime/performance_metrics/forecasting/probabilistic)

**Reply by fkiraly** [reply]
and in the tutorial here: https://github.com/sktime/sktime/blob/main/examples/01b_forecasting_proba.ipynb
